src/fpu/fpu_instructions.h

```diff
@@ -299,11 +299,12 @@ static void FPU_FBST(PhysPt addr) {
 
 	//numbers from back to front
 	for(Bitu i=0;i<9;i++){
-		Bit64u temp = rndint / 10;
-		Bit8u p = static_cast<Bit8u>(rndint % 10);
-		rndint = temp / 10;
-		p |= (static_cast<Bit8u>(temp % 10)) << 4;
-		mem_writeb(addr++,p);
+		const lldiv_t div10 = lldiv(rndint, 10);
```

Comment:
I think we need to assure, that `unsigned long long` is indeed the same type as `uint64_t`. Static assert before line 301? (You could replace `Bitu i` with simple `int i` at the same time, although it does not really matter - compiler will probably do it anyway while optimizing).

When playing with this code in Compiler Explorer I noticed, that if `mem_writeb` is moved out of the loop, the compiler nicely unrolls the loop (when compiled with -O3). Maybe it would result in faster code or maybe not (I can't say without doing microbenchmarks) - leaving it as an idea for future improvement, but I think it's out of the scope of this PR :).

Reply:
Added the static assert.

Reply:
> I noticed, that if mem_writeb is moved out of the loop, the compiler nicely unrolls the loop (when compiled with -O3). Maybe it would result in faster code or maybe not (I can't say without doing microbenchmarks) 

This means it's storing all 9 intermediate results from each loop pass; I guess we could store them in a temporary array, then write at the end (would be a tad uglier; but probably faster because it could SIMD the for loop with 18 calls to lldiv and stores).